.github/workflows/codeql.yml

```
@@ -0,0 +1,24 @@
+    security-events: write
+
+    # required to fetch internal or private CodeQL packs
+    packages: read
+
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
```

actionlint (1.7.7)

the runner of "github/codeql-action/init@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)